client/network/test/src/sync.rs

```diff
@@ -818,20 +820,25 @@ fn sync_to_tip_when_we_sync_together_with_multiple_peers() {
 
 	let block_hash =
 		net.peer(0)
-			.push_blocks_at_without_informing_sync(BlockId::Number(0), 10_000, false);
+			.push_blocks_at_without_informing_sync(BlockId::Number(0), 10_000, false, false);
+
+	let block_hash2 =
+		net.peer(1)
+			.push_blocks_at_without_informing_sync(BlockId::Number(0), 5_000, false, false);
 
-	net.peer(1)
-		.push_blocks_at_without_informing_sync(BlockId::Number(0), 5_000, false);
+	assert!(!net.peer(2).has_block(block_hash));
+	net.peer(0).network_service().new_best_block_imported(block_hash, 10_000);
 
 	net.block_until_connected();
 	net.block_until_idle();
 
-	assert!(!net.peer(2).has_block(block_hash));
-
-	net.peer(0).network_service().new_best_block_imported(block_hash, 10_000);
 	while !net.peer(2).has_block(block_hash) && !net.peer(1).has_block(block_hash) {
 		net.block_until_idle();
 	}
+
+	// verify that peer1 and peer3 also downloaded the other fork
+	assert!(net.peer(0).has_block(block_hash2));
```

Comment:
block_hash2 is not a fork. It is part of the same chain as block_hash

Reply:
You're right. I confused myself about how `push_block_without_informing_sync_works`